src/core/ext/filters/client_channel/resolver/dns/c_ares/dns_resolver_ares.cc

```diff
@@ -223,100 +223,88 @@ void AresDnsResolver::OnNextResolutionLocked(void* arg, grpc_error* error) {
   r->Unref(DEBUG_LOCATION, "next_resolution_timer");
 }
 
-bool ValueInJsonArray(grpc_json* array, const char* value) {
-  for (grpc_json* entry = array->child; entry != nullptr; entry = entry->next) {
-    if (entry->type == GRPC_JSON_STRING && strcmp(entry->value, value) == 0) {
+bool ValueInJsonArray(const Json::Array& array, const char* value) {
+  for (const Json& entry : array) {
+    if (entry.type() == Json::Type::STRING && entry.string_value() == value) {
       return true;
     }
   }
   return false;
 }
 
-char* ChooseServiceConfig(char* service_config_choice_json,
-                          grpc_error** error) {
-  grpc_json* choices_json = grpc_json_parse_string(service_config_choice_json);
-  if (choices_json == nullptr) {
-    *error = GRPC_ERROR_CREATE_FROM_STATIC_STRING(
-        "Service Config JSON Parsing, error: could not parse");
-    return nullptr;
-  }
-  if (choices_json->type != GRPC_JSON_ARRAY) {
+std::string ChooseServiceConfig(char* service_config_choice_json,
+                                grpc_error** error) {
+  Json json = Json::Parse(service_config_choice_json, error);
+  if (*error != GRPC_ERROR_NONE) return nullptr;
+  if (json.type() != Json::Type::ARRAY) {
     *error = GRPC_ERROR_CREATE_FROM_STATIC_STRING(
         "Service Config Choices, error: should be of type array");
     return nullptr;
   }
-  char* service_config = nullptr;
+  std::string service_config = nullptr;
   InlinedVector<grpc_error*, 4> error_list;
-  bool found_choice = false;  // have we found a choice?
```

Comment:
looks like there is a semantic change here. Earlier, we were validating all the choices, but now we choose the first choice and exit. The proposal suggests that we should be using the first choice that fits, so the new code seems right.

Reply:
Hmm, yeah, good point.  We probably should actually validate all of the options.  I've changed the code to preserve the old behavior.